index.js

```diff
@@ -508,8 +508,10 @@ RedisClient.prototype.connection_gone = function (why) {
     if (this.max_attempts && this.attempts >= this.max_attempts) {
         this.retry_timer = null;
         // TODO - some people need a "Redis is Broken mode" for future commands that errors immediately, and others
-        // want the program to exit.  Right now, we just log, which doesn't really help in either case.
+        // want the program to exit.  Right now, we just log, which doesn't really help in either case, and emit
+        // an error event.
         console.error("node_redis: Couldn't get Redis connection after " + this.max_attempts + " attempts.");
+        this.emit('error', new Error("Redis connection in broken state"));
```

Comment:
To distinguish this error from the other one it would be good to use different error messages. E.g.: Redis connection in broken state: maximum connection attempts exceeded." and "Redis connection in broken state: connection timeout exceeded.".

And please add a note in the readme about the emitted errors after exceeding the limit. Thx